includes/admin/feedzy-rss-feeds-admin.php

```diff
@@ -673,12 +673,25 @@ private function save_settings() {
 		$settings = apply_filters( 'feedzy_get_settings', array() );
 		switch ( $post_tab ) {
 			case 'general':
+				$auto_categories = isset( $_POST['auto-categories'] ) ? filter_input( INPUT_POST, 'auto-categories', FILTER_UNSAFE_RAW, FILTER_REQUIRE_ARRAY ) : array();
+
+				$auto_categories = array_filter( $auto_categories, function( $item ) {
+					if ( ! empty( $item['keywords'] ) && is_numeric( $item['category'] ) ) {
+						$item['keywords'] = sanitize_text_field( $item['keywords'] );
+						return true;
+					}
+					return false;
+				} );
```

Comment:
I think it is better to have a separate context here. A filter should filter the items, a map should modify the items.

```php
$auto_categories = array_filter( $auto_categories, function( $item ) {
    return ! empty( $item['keywords'] ) && is_numeric( $item['category'] );
});

$auto_categories = array_map(function( $item ) {
    $item['keywords'] = sanitize_text_field( $item['keywords'] );
    return $item;
}, $auto_categories);
```